src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/ServerNode.cs

```diff
@@ -70,6 +79,12 @@ internal SmoServerCreator ServerCreator
             }
         }
 
+        internal bool IsMasterConnection()

```

Comment:
we probably should check if the use is connecting to any of the system tables. e.g., master, tempdb, msdb or model.